src/com/mendix/widget/ProgressCircle/components/ProgressCircle.ts

```diff
@@ -0,0 +1,143 @@
+import { Component, DOM, createElement } from "react";
+
+import * as classNames from "classnames";
+import { Circle } from "progressbar.js";
+import { Alert } from "./Alert";
+
+import "../ui/ProgressCircle.css";
+
+export interface ProgressCircleProps {
+    animate?: boolean;
+    contextObject?: mendix.lib.MxObject;
+    maximumValue?: number;
+    microflow?: string;
+    onClickType?: ProgressOnclick;
+    page?: string;
+    pageSettings?: PageSettings;
+    textSize?: ProgressTextSize;
+    value: number | null;
+}
+
+export type ProgressTextSize = "small" | "medium" | "large";
+export type ProgressOnclick = "doNothing" | "showPage" | "callMicroflow";
+export type PageSettings = "content" | "popup" | "modal";
+
+export class ProgressCircle extends Component<ProgressCircleProps, { alertMessage: string }> {
+    static defaultProps: ProgressCircleProps = {
+        animate: true,
+        onClickType: "doNothing",
+        maximumValue: 100,
+        textSize: "medium",
+        value: 0
+    };
+    private progressNode: HTMLElement;
+    private progressCircle: Circle;
+
+    constructor(props: ProgressCircleProps) {
+        super(props);
+
+        this.state = { alertMessage: "" };
+    }
+
+    componentDidMount() {
+        this.checkConfig();
+        this.setProgress(this.props.value, this.props.maximumValue);
+    }
+
+    componentDidUpdate() {
+        this.setProgress(this.props.value, this.props.maximumValue);
+    }
+
+    render() {
+        return DOM.div({ className: "widget-progress-circle" },
+            DOM.div({
+                className: classNames(`widget-progress-circle-${this.props.textSize}`,
+                    {
+                        negative: this.props.value < 0,
+                        "red-progress-text": this.props.maximumValue < 1
+                    }
+                ),
+                onClick: () => this.handleOnClick(),
+                ref: node => this.progressNode = node
+            }),
+            this.state.alertMessage ? createElement(Alert, { message: this.state.alertMessage }) : null
```

Comment:
Not without turning it into a class or at least disrupting it's current tranquility as a pure stateless component.
Alternative is to always render it and control its visibility based on the `hidden` class. Don't think it's faithful to react to do it this way though.